### PR TITLE
CHANGES.md file for Chapel 1.21

### PR DESCRIPTION
**Pull request by bradcray**
Here is a short list of CHANGES that you might find present in Chapel 1.21

[entries contributed by the team, edited by me, reviewed by @ronawho]

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -1,6 +1,429 @@
...
+* added support for split initialization of variables, refs, types, and params
+  (see https://chapel-lang.org/docs/1.21/language/spec/variables.html#split-initialization)
+  (TODO language evolution link)
+* added support for copy elision transformations
+  (TODO: where to document?)
+* improved `out` intents to support split initialization and to reduce copies
+  (TODO language evolution link? spec?)
+* `record` temporaries may now be deinitialized at the end of a statement
+  (TODO: where to document?)
+* compile-time (`type`, `param`) methods now require `overload` when overloaded
```
Comment (cassella) [inline CHANGES.md]: ... now require `override`, I think.

File: CHANGES.md [hunk 2]
```diff
@@ -1,6 +1,429 @@
...
+  (see https://chapel-lang.org/docs/1.21/language/spec/statements.html#the-use-statement)
+* sub-modules no longer have lexical visibility into their parent modules
+  (see https://chapel-lang.org/docs/1.21/language/spec/modules.html#nested-modules)
+* the `locale` type now has value semantics and a default value of `Locales[0]`
+* values of type `string` are now UTF8-validated
+* `bytes.this` and `bytes.these` now return/yield `unit(8)` instead of `bytes`
+  (see https://chapel-lang.org/docs/1.21/builtins/Bytes.html)
+* stopped generating `init=` functions for records with a user-defined `init=`
+* an `init=` containing compilerError() now indicates the type is not copyable
+  (TODO: PR #14887 - not sure where this should be documented?)
```
Comment (mppf) [inline CHANGES.md]: I'm hoping @benharsh will handle this

File: CHANGES.md [hunk 3]
```diff
@@ -1,6 +1,429 @@
...
+* added support for split initialization of variables, refs, types, and params
+  (see https://chapel-lang.org/docs/1.21/language/spec/variables.html#split-initialization)
+  (TODO language evolution link)
```
Comment (mppf) [inline CHANGES.md]: https://chapel-lang.org/docs/1.21/language/evolution.html#split-initialization

File: CHANGES.md [hunk 4]
```diff
@@ -1,6 +1,429 @@
...
+* added support for copy elision transformations
+  (TODO: where to document?)
```
Comment (mppf) [inline CHANGES.md]: https://chapel-lang.org/docs/1.21/language/evolution.html#copy-elision
https://chapel-lang.org/docs/1.21/language/spec/variables.html#copy-elision

File: CHANGES.md [hunk 5]
```diff
@@ -1,6 +1,429 @@
...
+* improved `out` intents to support split initialization and to reduce copies
+  (TODO language evolution link? spec?)
```
Comment (mppf) [inline CHANGES.md]: https://chapel-lang.org/docs/master/language/spec/procedures.html#the-out-intent

File: CHANGES.md [hunk 6]
```diff
@@ -1,6 +1,429 @@
...
+* `record` temporaries may now be deinitialized at the end of a statement
+  (TODO: where to document?)
```
Comment (mppf) [inline CHANGES.md]: https://chapel-lang.org/docs/1.21/language/evolution.html#deinitialization-point-of-nested-call-expressions
https://chapel-lang.org/docs/1.21/language/spec/variables.html#deinit-points

File: CHANGES.md [hunk 7]
```diff
@@ -1,6 +1,429 @@
...
+* made the compiler no longer tolerate assignment overloads for classes
+
+New Features
+------------
+* added support for `import` statements as a more precise way of using modules
+  (see <TODO> doc link)
+* added support for renaming a module in its `use` statement
+  (see <TODO> doc link)
+* added explicit relative `import` and `use` chains via `this.` and `super.`
+  (TODO: where to document?)
```
Comment (mppf) [inline CHANGES.md]: https://chapel-lang.org/docs/master/language/spec/statements.html#the-use-statement

File: CHANGES.md [hunk 8]
```diff
@@ -1,6 +1,429 @@
...
+* implemented prototypical support for storing submodules in different files
+  (TODO: where to document?)
```
Comment (mppf) [inline CHANGES.md]: https://chapel-lang.org/docs/master/technotes/module_include.html

File: CHANGES.md (hunk 1 above)
Comment (bradcray) [reply CHANGES.md]: Oops, thank you!

File: CHANGES.md [hunk 9]
```diff
@@ -1,6 +1,481 @@
...
+* generally updated documentation with respect to the changes in this release
+* improved the documentation for `owned` and `shared` classes
+  (see https://chapel-lang.org/docs/1.21/builtins/OwnedObject.html and
+   https://chapel-lang.org/docs/1.21/builtins/SharedObject.html)
+* improved the description of tuple semantics in the language specification
+  (see https://chapel-lang.org/docs/1.21/language/spec/tuples.html#value-tuples-and-referential-tuples)
+* increased usage of the `:throws:` tag in library documentation
+* added details about special I/O methods to the forwarding technote
+  (see https://chapel-lang.org/docs/1.21/technotes/forwarding.html#resolving-forwarded-methods)
+
```
Comment (ben-albrecht) [inline CHANGES.md]: Recent change (just merged):

```suggestion
* improved the documentation of the `UnitTest` module
  (see https://chapel-lang.org/docs/1.21/modules/packages/UnitTest.html)
  
```

**Pull request merged**